docs/technical-design.md

```diff
@@ -0,0 +1,130 @@
+# Technical Design
+
+## To be discussed
+
+
+## Istio operator
+
+![IstioOperator Overview](./istio-operator-overview.svg)
+
+### Ownership of current resources in Kyma repository
+
+In the transitioning to a more modularised architecture, the [IstioOperator resource](https://github.com/kyma-project/kyma/tree/main/resources/istio), 
+the [additional istio-resources](https://github.com/kyma-project/kyma/tree/main/resources/istio-resources) and 
+the [certificates](https://github.com/kyma-project/kyma/tree/main/resources/certificates) must be moved to the new modules.
+
+#### IstioOperator resource
+This is moved into our new Istio operator and used to define the default values for Istio, which can be customised by the user through `IstioCR`.
+
+#### istio-resources
+
+##### Istio Grafana dashboards
+It's still needs to be decided who will have the ownership of the dashboards. There are different things to consider like the change interval or relevance of Istio version updates.
+
+##### Istio ServiceMonitor & istio-healthz Virtual Service
+They provide monitoring capability for the Istio installation and should therefore be reconciled by the operator.
```

Comment:
I'm planning to replace the direct dependency by a simple annotation: https://github.com/kyma-project/kyma/pull/16247